flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerNG.java

```diff
@@ -125,7 +127,8 @@ void notifyKvStateUnregistered(
     CompletableFuture<String> triggerSavepoint(
             @Nullable String targetDirectory, boolean cancelJob, SavepointFormatType formatType);
 
-    CompletableFuture<String> triggerCheckpoint();
+    CompletableFuture<CompletedCheckpoint> triggerCheckpoint(
+            @Nullable CheckpointType checkpointType);
```

Comment:
Is this actually nullable? Seems strange give that its not nullable in the request body.

Reply:
Why cant we have the MiniCluster default to `CheckpointType#DEFAULT`, or whatever was set as the default based on the configuration.

Reply:
Good point. Let me see if I can do this without breaking any public API